src/client/MenuWindow.cpp

```diff
@@ -2,186 +2,372 @@
         {
-            std::cerr << "Error while opening json ressources file /img/menu/menu.json" << std::endl;
-            exit(1);
+            button.delChar();
         }
-        std::string str((std::istreambuf_iterator<char>(file)), std::istreambuf_iterator<char>());
+    }
+}
 
-        std::unique_ptr<Json::CharReader> reader = std::unique_ptr<Json::CharReader>(Json::CharReaderBuilder().newCharReader());
-        Json::Value obj;
-        std::string errors;
-        reader->parse(str.c_str(), str.c_str() + str.size(), &obj, &errors);
+/*!
+ * \brief Load all the textures that will be display on the menu
+ */
+void MenuWindow::loadMenuTexture()
+{
 
-        const Json::Value &data = obj["data"];
+    backgroundTexture = (std::unique_ptr<TextureDisplayer>)new TextureDisplayer(RESOURCES_PATH "/img/menu/background.png");
+    backgroundTexture->addSprite();
+    float backgroundScale = 1 / (float(backgroundTexture->getWidth()) / float(gameEnginePtr->clientWindow->getSize().x ));
+    backgroundTexture->setHudSpritePosition(backgroundScale, gameEnginePtr->clientWindow->getSize().x , gameEnginePtr->clientWindow->getSize().y , 0, 0);
 
-        for (unsigned index = 0; index < data.size(); ++index)
-        {
-            menuButtons.emplace_back(   sf::Vector2f(data[index]["width"].asFloat() * gameTitle->getLocalBounds().width, data[index]["height"].asFloat() * gameTitle->getLocalBounds().height), 
-                                        sf::Vector2f(   gameTitle->getPosition().x + data[index]["x"].asFloat() * gameTitle->getLocalBounds().width, 
-                                                        gameTitle->getPosition().y - data[index]["y"].asFloat() * gameTitle->getLocalBounds().height), buttonColor);
-            menuButtons.back().setText(40, sf::Vector2f(0, 0), data[index]["text"].asString(), *menuFont);
-        }
+    if (!menuFont.loadFromFile(RESOURCES_PATH "/img/hud/font.otf"))
+    {
+        std::cerr << "Font not loaded" << std::endl;
     }
 
-    /*!
-     * @brief Function that deteck where the user click and what to send to the engine
-     * @param timeSecond is a boolean used to 
-     */
-    long MenuWindow::getCurrentTime(bool timeSecond)
+    std::ifstream file(RESOURCES_PATH "/img/menu/menu.json");
+    if (!file.is_open())
     {
-        if (timeSecond)
-        {
-            return std::chrono::duration_cast<std::chrono::seconds>(std::chrono::system_clock::now().time_since_epoch()).count();
-        } 
-        else 
+        std::cerr << "Error while opening json ressources file /img/menu/menu.json" << std::endl;
+        exit(1);
+    }
+    std::string str((std::istreambuf_iterator<char>(file)), std::istreambuf_iterator<char>());
+
+    std::unique_ptr<Json::CharReader> reader = std::unique_ptr<Json::CharReader>(Json::CharReaderBuilder().newCharReader());
+    Json::Value obj;
+    std::string errors;
+    reader->parse(str.c_str(), str.c_str() + str.size(), &obj, &errors);
```

Comment:
Don't we have a function to load Json file ? If it's not in this class you can also call gameEnginePtr->load or something like that.